src/mock_display.rs

```diff
@@ -563,12 +633,195 @@ where
     }
 }
 
+struct FancyPanic<'a, C>
+where
+    C: PixelColor,
+{
+    display: &'a MockDisplay<C>,
+    expected: &'a MockDisplay<C>,
+}
+
+impl<'a, C> FancyPanic<'a, C>
+where
+    C: PixelColor,
+{
+    fn new(display: &'a MockDisplay<C>, expected: &'a MockDisplay<C>) -> Self {
+        Self { display, expected }
+    }
+}
+
+fn write_display<C>(
+    f: &mut fmt::Formatter<'_>,
+    display: &MockDisplay<C>,
+    bounding_box: &Rectangle,
+) -> fmt::Result
+where
+    C: PixelColor + ColorMapping,
+{
+    for y in bounding_box.rows() {
+        if y % 2 != 0 {
+            // Skip all odd y coordinates, because `write_row` outputs two pixel rows with
+            // a single row of characters.
+            continue;
+        }
```

Comment:
I think this has the same behaviour, no?

```suggestion
    for y in bounding_box.rows().step_by(2) {
```

Reply:
Thanks for the suggestion, that's better. I was too lazy to look up a better solution when I wrote the code.